tests/components/homewizard/test_init.py

```diff
@@ -171,3 +174,92 @@ async def test_load_handles_initialization_error(aioclient_mock, hass):
     await hass.async_block_till_done()
 
     assert entry.state is ConfigEntryState.SETUP_RETRY or ConfigEntryState.SETUP_ERROR
+
+
+async def test_coordinator_sets_update_interval(aioclient_mock, hass):
+    """Test coordinator calculates correct update interval."""
+
+    device = get_mock_device()
+
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={CONF_IP_ADDRESS: "1.1.1.1"},
+        unique_id=DOMAIN,
+    )
+    entry.add_to_hass(hass)
+
+    with patch(
+        "aiohwenergy.HomeWizardEnergy",
+        return_value=device,
+    ):
+        await hass.config_entries.async_setup(entry.entry_id)
+
+    await hass.async_block_till_done()
+
+    assert hass.data[DOMAIN][entry.entry_id][
```

Comment:
Please don't assert things directly on the coordinator. If we want to test the update interval and coordinator updates we can move time forward and check that another call was made to the library mock.

Here's our test helper for moving time forward:
https://github.com/home-assistant/core/blob/240c9979c7e7390a18257fcfacf208a43f7ce08c/tests/common.py#L379-L383